### PR TITLE
[ngtcp2] Update to v1.9.0

### PR DESCRIPTION
**Pull request by donny-dont**
- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [X] SHA512s are updated for each updated download.
- [X] The "supports" clause reflects platforms that may be fixed by this new version.
- [X] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [X] Any patches that are no longer applied are deleted from the port's directory.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/ngtcp2/portfile.cmake [hunk 1]
```diff
@@ -1,11 +1,16 @@
-vcpkg_from_github(
+set(FILENAME "ngtcp2-${VERSION}.tar.xz")
+
+vcpkg_download_distfile(ARCHIVE
+    URLS "https://github.com/ngtcp2/ngtcp2/releases/download/v${VERSION}/ngtcp2-${VERSION}.tar.xz"
+    FILENAME "ngtcp2-${VERSION}.tar.xz"
+    SHA512 2ecec2ec968878caa41df621a9054cca00b1b7b96dedc7c9c6f46b3c5e7534d9b07c985bdfc5e762ee68751e40ba06318aa54948ec01219eee4156f717e85b8c
+)
```
Comment (donny-dont) [inline ports/ngtcp2/portfile.cmake:7]: The library authors recommend downloading the tar balls. https://github.com/ngtcp2/ngtcp2/releases/tag/v1.9.0

File: ports/ngtcp2/openssl_required.patch [hunk 2]
```diff
@@ -1,14 +0,0 @@
-diff --git a/CMakeLists.txt b/CMakeLists.txt
```
Comment (donny-dont) [inline ports/ngtcp2/openssl_required.patch:1]: Obsoleted by https://github.com/ngtcp2/ngtcp2/commit/5f9547e96819b10a65723a0cc3137be263a714f2

**Review by donny-dont — COMMENTED** [review]
🤞 this is a quick one 😄 

I did check that things built with all the different features on `x64-windows` so I think this is pretty good to go

File: ports/ngtcp2/portfile.cmake (hunk 1 above)
Comment (dg0yt) [reply ports/ngtcp2/portfile.cmake:7]: And what is the difference, apart from `autoreconf` and `ChangeLog`?

**Comment by JonLiu1993** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

File: ports/ngtcp2/portfile.cmake (hunk 1 above)
Comment (donny-dont) [reply ports/ngtcp2/portfile.cmake:7]: This got brought up in an issue and here is the maintainers rationale

https://github.com/ngtcp2/ngtcp2/issues/1421#issuecomment-2485845334

Comment (dg0yt) [reply ports/ngtcp2/portfile.cmake:7]: The git submodules are probably neither wanted (vendored deps) nor needed (examples and tests) in the vcpkg port.

Comment (donny-dont) [reply ports/ngtcp2/portfile.cmake:7]: Ok so you want this part reverted and to use the `vcpkg_from_github`.

File: ports/ngtcp2/openssl_required.patch
Comment (talregev) [inline ports/ngtcp2/openssl_required.patch:1]: Please don't delete this.
The ngtcp2 can compile without openssl.
And quiet detection already lead to bug in vcpkg that we fix.

File: ports/ngtcp2/portfile.cmake (hunk 1 above)
Comment (talregev) [reply ports/ngtcp2/portfile.cmake:7]: I also in favor to switch vcpkg& from_git

File: ports/ngtcp2/openssl_required.patch
Comment (donny-dont) [reply ports/ngtcp2/openssl_required.patch:1]: Hi @talregev ! I’m the author of the PR that fixed this issue. Please see https://github.com/ngtcp2/ngtcp2/pull/1385 which explains the behavior.

File: ports/ngtcp2/portfile.cmake (hunk 1 above)
Comment (dg0yt) [reply ports/ngtcp2/portfile.cmake:7]: `vcpkg_from_github` (SHA512-cacheable, supporting `--head`), not `vcpkg_from_git` (cacheable, not verified).

File: ports/ngtcp2/portfile.cmake [hunk 3]
```diff
@@ -1,11 +1,16 @@
...
+
+# Extract archive
+vcpkg_extract_source_archive_ex(
```
Comment (dg0yt) [inline ports/ngtcp2/portfile.cmake:10]: The issue of this function being deprecated goes away when restoring vcpkg_from_github.
And no redundant comments, please.

File: ports/ngtcp2/openssl_required.patch
Comment (talregev) [reply ports/ngtcp2/openssl_required.patch:1]: Thank you for link.
How it react with libressl after your change?

File: ports/ngtcp2/portfile.cmake (hunk 1 above)
Comment (talregev) [reply ports/ngtcp2/portfile.cmake:7]: I was writing from my phone.
It was typo. I meant for `vcpkg_from_github`

**Comment by JonLiu1993** [comment]
Tested features `libressl,wolfssl` successfully in the following triplet:
- x86-windows
- x64-windows
- x64-windows-static

File: ports/ngtcp2/openssl_required.patch
Comment (donny-dont) [reply ports/ngtcp2/openssl_required.patch:1]: Everything worked great. We use libressl for our TLS on the Windows WebKit port.

**Review by JonLiu1993 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the update and sorry for the build lab instability!

**Pull request merged**